modules/openid_connect/lib/open_project/openid_connect/session_mapper.rb

```diff
@@ -0,0 +1,77 @@
+module OpenProject::OpenIDConnect
+  class SessionMapper
+    def self.handle_logout(logout_token)
+      link = ::OpenIDConnect::UserSessionLink.find_by(oidc_session: logout_token.sid)
+      new(link).expire!
+    rescue StandardError => e
+      Rails.logger.error { "Failed to handle OIDC session logout: #{e.message}" }
```

Comment:
The error should not be swallowed but re-raised so that the omniauth responds with HTTP 400 if the logout somehow failed instead of 200 as per the specification.

Reply:
Now the question is what we do if a link is found but the linked session isn't there anymore.
One could argue that we should not throw an error because the user is not logged in anymore, so the logout has been successful.

But then you could also argue that not finding the link is not an error either, because due to the cascade policy that ought to mean that the session is gone too and the user logged out as well.

Reply:
I've pushed the re-raise of the error, I haven't found an exact reference to your second behavior in the spec. I'd not throw an error for now until we have some experience with how the errors are handled on the idP side.